airbyte-cdk/python/airbyte_cdk/sources/declarative/manifest_declarative_source.py

```diff
@@ -175,33 +125,18 @@ def _configure_logger_level(self, logger: logging.Logger):
             logger.setLevel(logging.DEBUG)
 
     def _validate_source(self):
-        # Validates the connector manifest against the schema auto-generated from the low-code backend
-        full_config = {}
-        if "version" in self._legacy_source_config:
-            full_config["version"] = self._legacy_source_config["version"]
-        full_config["check"] = self._legacy_source_config["check"]
-        streams = [
-            self._legacy_factory.create_component(stream_config, {}, False)()
-            for stream_config in self._stream_configs(self._legacy_source_config)
-        ]
-        if len(streams) > 0:
-            full_config["streams"] = streams
-        declarative_source_schema = ConcreteDeclarativeSource.json_schema()
-
-        try:
-            validate(full_config, declarative_source_schema)
-        except ValidationError as e:
-            raise ValidationError("Validation against auto-generated schema failed") from e
-
-        # Validates the connector manifest against the low-code component json schema
         try:
             raw_component_schema = pkgutil.get_data("airbyte_cdk", "sources/declarative/declarative_component_schema.yaml")
             declarative_component_schema = yaml.load(raw_component_schema, Loader=yaml.SafeLoader)
         except FileNotFoundError as e:
             raise FileNotFoundError(f"Failed to read manifest component json schema required for validation: {e}")
 
+        streams = self._source_config.get("streams")
```

Comment:
Instead of this additional validation, it might be better to document in the schema YAML and rely on the validator to catch manifests that don't specify at least one stream. wdyt?
```
title: DeclarativeSource
type: object
...
properties:
  type:
    type: string
    enum: [DeclarativeSource]
  streams:
    type: array
    items:
      "$ref": "#/definitions/DeclarativeStream"
    minItems: 1
```

Reply:
I think [this](https://github.com/airbytehq/airbyte/pull/20747#discussion_r1057401169) is still relevant and prevents us from doing this easily with the auto-generated schemas since [the solution provided by pydantic](https://docs.pydantic.dev/usage/schema/#unenforced-field-constraints) does not align with autogeneration.

Reply:
ah okay yeah we can leave as is